src/core/sync/mutex.d

```diff
@@ -372,6 +372,9 @@ unittest
     // by checking that locking is not possible. This assumes
     // that the underlying implementation is well behaved
     // and makes the object non-lockable upon destruction.
+    // For example, Bionic doesn't appear to do so, so this test is
+    // not run on Android.
+    version (CRuntime_Bionic) {} else
```

Comment:
This test asserts on Android 6.0 Marshmallow, as [it doesn't appear that `pthread_mutex_destroy` really destroys the mutex there](https://android.googlesource.com/platform/bionic/+/marshmallow-release/libc/bionic/pthread_mutex.cpp#639).  I notice that if I comment out the line that sets `PTHREAD_MUTEX_RECURSIVE` in this druntime module, the test passes again, but I haven't dug into why it only doesn't work for recursive mutexes.  It doesn't, so this test can't be run as is.